### DISCUSSION TITLE
EntityId component definition --> require concept or components?

### DISCUSSION THREAD

**Discussion by droqen** [AmbientRun/Ambient#1083] [Ideas]
Writing the `ambient.toml` for a 'flag' concept, I opened by writing this line:
```
[component]
flagpole = { type = "EntityId" } # flagpole must have translation()
```
It occurred to me that it would be very useful to have a way to require, or even just expect/suggest, certain properties of the pointed-to entity. I'm not sure how I'd like it formatted, but here's a couple first drafts with not much foresight:
```
[component]
flagpole = { type = "EntityId", require_components = ["translation"] }
```
```
[component]
flagpole = { type = "EntityId", require_concepts = ["transformable"] }
```

**Comment by droqen** [comment]
Topic related (ha ha) to relational queries https://github.com/AmbientRun/Ambient/discussions/600

**Comment by FredrikNoren** [comment]
Yup! Been thinking about this also, would be very handy (especially if the engine can guarantee this relationship).
